src/components-examples/material/tabs/tab-group-harness/tab-group-harness-example.spec.ts

```diff
@@ -0,0 +1,61 @@
+import {TestBed, ComponentFixture, waitForAsync} from '@angular/core/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
+import {MatTabGroupHarness} from '@angular/material/tabs/testing';
+import {HarnessLoader} from '@angular/cdk/testing';
+import {BrowserDynamicTestingModule, platformBrowserDynamicTesting}
+  from '@angular/platform-browser-dynamic/testing';
+import {MatTabsModule} from '@angular/material/tabs';
+import {TabGroupHarnessExample} from './tab-group-harness-example';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+
+describe('TabGroupHarnessExample', () => {
+  let fixture: ComponentFixture<TabGroupHarnessExample>;
+  let loader: HarnessLoader;
+
+  beforeAll(() => {
+    TestBed.initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting());
+  });
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [MatTabsModule, NoopAnimationsModule],
+        declarations: [TabGroupHarnessExample]
+      }).compileComponents();
+      fixture = TestBed.createComponent(TabGroupHarnessExample);
+      fixture.detectChanges();
+      loader = TestbedHarnessEnvironment.loader(fixture);
+    })
+  );
+
+  it('should load harness for tab-group', async () => {
+    const tabGroups = await loader.getAllHarnesses(MatTabGroupHarness);
+    expect(tabGroups.length).toBe(1);
+  });
+
+  it('should load harness for tab-group with selected tab label', async () => {
+    const tabGroups = await loader.getAllHarnesses(MatTabGroupHarness.with({
+      selectedTabLabel: 'First',
+    }));
+    expect(tabGroups.length).toBe(1);
+  });
+
+  it('should be able to get tabs of tab-group', async () => {
+    const tabGroup = await loader.getHarness(MatTabGroupHarness);
+    const tabs = await tabGroup.getTabs();
+    expect(tabs.length).toBe(3);
+  });
+
+  it('should be able to select tab from tab-group', async () => {
+    const tabGroup = await loader.getHarness(MatTabGroupHarness);
+    expect(await (await tabGroup.getSelectedTab()).getLabel()).toBe('First');
+    await tabGroup.selectTab({label: 'Second'});
+    expect(await (await tabGroup.getSelectedTab()).getLabel()).toBe('Second');
+  });
+
+  it('should throw error when attempting to select invalid tab', async () => {
```

Comment:
This is a test for our internal implementation, I don't think it'll be useful for somebody consuming the tabs module.